vcf.c

```diff
@@ -2901,6 +2901,22 @@ int vcf_parse(kstring_t *s, const bcf_hdr_t *h, bcf1_t *v)
     return ret;
 }
 
+int vcf_open_mode(char *mode, const char *fn, const char *format)
+{
+    if (format == NULL) {
+        // Try to pick a format based on the filename extension
+        char extension[7];
+        if (find_file_extension(fn, extension) < 0) return -1;
+        return vcf_open_mode(mode, fn, extension);
+    }
+    else if (strcmp(format, "bcf") == 0) strcpy(mode, "b");
+    else if (strcmp(format, "vcf") == 0) strcpy(mode, "");
+    else if (strcmp(format, "vcf.gz") == 0) strcpy(mode, "z");
```

Comment:
Another extension often used is  `vcf.bgz` and `strcasecmp` to make the comparison case insensitive.